src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexFindOptimizations.cs

```diff
@@ -271,7 +276,7 @@ private static (string String, int Distance)? FindFixedDistanceString(List<Fixed
                 for (int i = 0; i < fixedDistanceSets.Count + 1; i++)
                 {
                     char[]? chars = i < fixedDistanceSets.Count ? fixedDistanceSets[i].Chars : null;
-                    bool invalidChars = chars is not { Length: 1 };
+                    bool invalidChars = chars is not { Length: 1 } || fixedDistanceSets[i].Negated;
```

Comment:
Can you run the FunctionalTests with /p:Coverage=true and make sure all the new code paths are covered (and if not, add tests to ensure they are)?

Reply:
The only places I see missing any coverage are the negated Chars FixedDistanceSets in `SortFixedDistanceSetsByQuality`.

Specifically
https://github.com/dotnet/runtime/blob/981d33c2ae465f9156f85817763d3926f5c6f3a7/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexPrefixAnalyzer.cs#L457-L465
And
https://github.com/dotnet/runtime/blob/981d33c2ae465f9156f85817763d3926f5c6f3a7/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexPrefixAnalyzer.cs#L471-L474

I failed to come up with a pattern that would hit those paths (presumably it should contain multiple length=1 negated `FixedDistanceSet`s).
Any tips on what those might look like appreciated :)

Reply:
That's building up a list of all possible characters that could occur at a specific distance from the start of the pattern.  Alternations are your best bet.  So, for example, if you have the pattern:
```
Steve|Miha
```
that's going to generate the sets:
```
[SM]
[ti]
[eh]
[va]
```
or if you have:
```
[0-9][a-z]|[A-Z]7
```
that'll generate the sets:
```
[0-9A-Z]
[a-z7]
```

Reply:
Oh, wait, those are for negated-related checks. I'd need to look at the code, but it might not ever create such sets.

Reply:
The most I could get it to do was a single length=1 negated set (e.g. literally `[^a]`) which is likely coming from the `FindFirstCharClass` fallback if we found nothing useful.

It does produce negated sets for patterns like `[a-q][^u-z]{13}x` that will turn into
`[a-q], [^u-z], [^u-z], [^u-z], [^u-z], [^u-z], ...`
I just wasn't sure if it's possible for length=1 sets as well.

This PR doesn't affect the kinds of sets we initially produce, but it will change sets that would previously become `Range` with `low==high` into Length=1 `Chars`.
I initially did this to use `Chars` for length 1-5 negated sets instead of `IndexOfAnyValues`, but since then I've reverted the negated `IndexOfAnyValues` from this path. Now some of this new logic is useless and it should just defer to `Range`, I'll clean this up.

Reply:
Removed the unnecessary logic I introduced with this PR, code coverage now looks good.